cmd/boulder-va/main.go

```diff
@@ -27,7 +27,8 @@ type Config struct {
 		// before giving up. May be short-circuited by deadlines. A zero value
 		// will be turned into 1.
 		DNSTries                  int
-		DNSProvider               *cmd.DNSProvider `validate:"required"`
+		DNSProvider               *cmd.DNSProvider `validate:"required_without=DNSStaticResolvers"`
+		DNSStaticResolvers        []string         `validate:"required_without=DNSProvider"`
```

Comment:
One additional validation here. We use the same validation for `IP:Port` pairs in `redis.Config.ShardAddrs`.

```suggestion
		DNSStaticResolvers        []string         `validate:"required_without=DNSProvider,dive,hostname_port"`
```

Reply:
thanks!  What does "dive" do?